### DISCUSSION TITLE
Cannot retrieve Link's props from createLocalizedPathnamesNavigation

### DISCUSSION THREAD

**Discussion by JulienRst** [amannn/next-intl#1265] [Q&A] [answered]
### Description

Hello everyone, 

I need a `<Link />` component with special things (like analytics 👻 ), so I figured that I should create another link let's say `<LinkWithAnalytics />` that would forward ref to `<Link />`.

Let's keep it simple : 

```tsx
const { Link, redirect, usePathname, useRouter } =
  createLocalizedPathnamesNavigation({
    locales: ['en', 'fr'],
    localePrefix: 'as-needed',
    pathnames,
  });

const AnalyticLinks = React.forwardRef<
  React.ElementRef<typeof Link>, 
  React.ComponentPropsWithoutRef<typeof Link>
>(props, ref) => {
    return <Link ref={ref} ...props  /> // <-- error here
} 
```
> Property 'href' is missing in type '{ ref: ForwardedRef<never>; }'

So I just tried to check what is inside `typeof Link` and it's a bit hard to read but eventually result in `{}`.

It mostly related to HOW this is typed, because when used directly the Link have the correct props AND typed `href` from `pathnames`. 

And we're not safe from an error on my part 🤭 

If I can be of any form of help do not hesitate to reach out!

Kind regards 😄 

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

https://github.com/JulienRst/next-intl-bug-repro-app-router

### Reproduction description

1. Open editor
2. Check `navigation.ts` and acknowledge the uses of `createLocalizedPathnamesNavigation`
3. Check `analytics-link.tsx` and acknowledge the error

### Expected behaviour

As I was working with `createSharedPathnamesNavigation` before, I would expect that `typeof Link` give us the same props (with the href typed from `pathnames`) than the `typeof Link` from `createSharedPathnamesNavigation`.

**Comment by amannn** [comment, marked as answer]
Please see [How can I compose the link with its href prop?](https://next-intl-docs.vercel.app/docs/routing/navigation#link-composition) in the docs for `Link` (make sure you have "Localized pathnames" selected).

Going to move this to a discussion since it's a usage question and not a bug report.

**Comment by JulienRst** [comment]
Hey there, thanks for your answer, it really helped me out !
